src/vehicle.cpp

```diff
@@ -5182,11 +5182,14 @@ bool vehicle::fire_turret (int p, bool /* burst */ )
         if( liquid_fuel < 1 || charges < 1 ) {
             return false;
         }
-        itype *am_type = item::find_type( amt );
+        itype *am_type = item::find_type( default_ammo( amt ) );
+        itype *fake;
+        bool fake_ammo = false;
         if( !am_type->ammo ) {
-            // Let non-ammo-typed ammo be used too
-            // 0 in all stats, depends on gun to provide everything
-            am_type = item::find_type( "fake_ammo" );
+            // Alter ammo type temporarily
+            fake = item::find_type( "fake_ammo" );
+            std::swap( am_type->ammo, fake->ammo );
```

Comment:
Why would you need a dual type gun? The gun fires water, whether it is mounted or not. Please explain why you need dual typed items. Example for the new system:
```
"type": "GENERIC",
"volume": ... generic data, applies to all item types ...,
"ammo_data": {
    "damage": ... only ammo specific data here ...
},
"gun_data": {
    "range": ... only gun specific data here ...
}
"armor_data": {
    "warmth": ... only armor specific data here ...
}
```
If a armor_data entry exists, the item is armor. If a gun_data entry exists, it's a gun etc.
The item above would be a gun, but also be usable as ammo. This combination would probably cause some problems because ammo is counted by charges, but charges are also used to store the number of ammo items the gun is loaded with.
The base type can also be COMESTIBLE or ARMOR or GUN etc. that is still accepted because I didn't wanted to change all the existing json data.


For water as ammo this might do:
```
"type": "COMESTIBLE",
... existing water properties ...
"ammo_data": {
    "damage": ... specific data for usage as ammo here ...
}
```